dataland-csvconverter/pdfMapping.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+set -euxo pipefail
+# Utility script to generate a mapping between a folder of pdf reports and the ISINs in the real data set. Expects two
+# inputs: the csv file containing the real data and the folder with the pdfs for mapping. The output is a folder "pdfs"
+# containing all relevant pdfs and the mapping file. The latter can be used to update the real data set.
+# In order to check which pdfs do not comply with the expected format run:
+# ls *.pdf | grep -Pv '_[A-Z0-9]{12}_(Annual|Sustainability|Integrated).*Report'
+# inside the folder containing the pdfs.
+
+csv_file="$1"
+pdf_folder="$2"
+
+if [[ ! -f "$csv_file" ]]; then
+  echo "Error: Expected file $csv_file does not exist."
+  exit 1
+fi
+
+output_folder=$(dirname "$0")/pdfs
+mkdir -p "$output_folder"
+rm "$output_folder"/*.pdf || true
+output_file=$output_folder/output.csv
+
+echo "Annual Report File,Sustainability Report File,Integrated Report File,Annual Report,Sustainability Report,Integrated Report" > "$output_file"
+sed 1d "$csv_file" | while read -r line; do
+  isin=$(echo "$line" | grep -oP ';([A-Z0-9]{12});' | cut -d';' -f2) || true
+  if [[ -z "$isin" ]]; then
```

Comment:
If i understood it correctly, if a company does not have an ISIN this script does not work, as there is no other way to connect the report to the company entry. So in the future this might has to change when other identifiers are also used. Maybe as a loop through all existing identifiers of a company and check for every ID if there is a report corresponding to it.